lamindb/_artifact.py

```diff
@@ -1071,8 +1071,17 @@ def delete(
             storage = False
         # only delete in storage if DB delete is successful
         # DB delete might error because of a foreign key constraint violated etc.
-        self._delete_skip_storage()
-        if self.key is None or self._key_is_virtual:
+        if self._overwrite_versions and self.is_latest:
+            # includes self
+            for version in self.versions.all():
+                _delete_skip_storage(version)
+        else:
+            self._delete_skip_storage()
+        # by default do not delete storage if deleting only a previous version
+        # and the underlying store is mutable
+        if self._overwrite_versions and not self.is_latest:
+            delete_in_storage = False if storage is None else storage
```

Comment:
Do we want to allow forcing a storage delete here? I don't think so.

I'd issue a warning and say "storage argument is ignored; can't delete store on a previous version".

Otherwise the user is left with a family of records that doesn't have any storage attached to it. That's not useful, is it?

Reply:
Normally storage should not be specified at all, and it is, i assume a good reason to do this.

Reply:
I also want `storage` to be fully consistent for all cases, while we have it.

Reply:
If you don't want the warning, how about throwing an error? I can't imagine a single use case where you'd want to produce dangling records in that version family.

It's only going to be called by an inexperienced user who will produce a mess with this. Don't you agree?

Reply:
For edge-case cleanups with corrupted version chain or something like this.

Reply:
> a corrupted version chain or something like this

In that case the user should get an error telling them to delete the latest version so that the whole thing gets cleaned up.

You definitely don't want to keep around the corrupted version chain (or a similar case). Are you not afraid of the messiness this call will leave behind?

Reply:
Ok, after some think i agree that you are right here, added ignore and a warning.